### PR TITLE
update docs to use SystemJS object

### LINKED ISSUE #1500: Update docs to use SystemJS object
The docs still use the old API.

Example: https://github.com/systemjs/systemjs/blob/master/docs/system-api.md

### PR DESCRIPTION
**Pull request by b-strauss**
closes #1500

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ SystemJS
 Universal dynamic module loader - loads ES6 modules, AMD, CommonJS and global scripts in the browser and NodeJS.
 
 * [Loads any module format](docs/module-formats.md) with [exact circular reference and binding support](https://github.com/ModuleLoader/es6-module-loader/blob/v0.17.0/docs/circular-references-bindings.md).
-* Loads [ES6 modules compiled into the `System.register` bundle format for production](docs/production-workflows.md), maintaining circular references support.
+* Loads [ES6 modules compiled into the `SystemJS.register` bundle format for production](docs/production-workflows.md), maintaining circular references support.
```
Comment (guybedford) [inline README.md]: System.register is still called System.register, despite SystemJS naming. I know this is somewhat confusing.

File: README.md [hunk 2]
```diff
@@ -48,7 +48,7 @@ Basic Use
 </script>
 ```
 
-The above will support loading all module formats, including ES Modules transpiled into the System.register format.
+The above will support loading all module formats, including ES Modules transpiled into the SystemJS.register format.
```
Comment (guybedford) [inline README.md]: System.register

File: docs/config-api.md [hunk 3]
```diff
@@ -55,18 +55,18 @@ Bundles allow a collection of modules to be downloaded together as a package whe
...
 });
 ```
 
-In the above any require to `dependencyA` or `dependencyB` will first trigger a `System.import('bundleA')` before proceeding with the load of `dependencyA` or `dependencyB`.
+In the above any require to `dependencyA` or `dependencyB` will first trigger a `SystemJS.import('bundleA')` before proceeding with the load of `dependencyA` or `dependencyB`.
 
 It is an alternative to including a script tag for a bundle in the page, useful for bundles that load dynamically where we want to trigger the bundle load automatically only when needed.
 
-The bundle itself is a module which contains named System.register and define calls as an output of the builder. The dependency names the bundles config lists should be the same names that are explicitly defined in the bundle.
+The bundle itself is a module which contains named SystemJS.register and define calls as an output of the builder. The dependency names the bundles config lists should be the same names that are explicitly defined in the bundle.
```
Comment (guybedford) [inline docs/config-api.md]: System.register

File: docs/module-formats.md [hunk 4]
```diff
@@ -6,12 +6,12 @@ The following module formats are supported:
 * `cjs`: [CommonJS](#commonjs)
 * `amd`: [Asynchronous Module Definition](#amd)
 * `global`: [Global shim module format](#globals)
-* `register`: [System.register](system-api.md#systemregister-name-deps-declare) or [System.registerDynamic](system-api.md#systemregisterdynamic-name-deps-executingrequire-declare) compatibility module format
+* `register`: [SystemJS.register](system-api.md#systemregister-name-deps-declare) or [SystemJS.registerDynamic](system-api.md#systemregisterdynamic-name-deps-executingrequire-declare) compatibility module format
```
Comment (guybedford) [inline docs/module-formats.md]: System.register, System.registerDynamic

File: docs/module-formats.md [hunk 5]
```diff
@@ -26,8 +26,8 @@ When the module format is not set, automatic regular-expression-based detection
 This module format detection is never completely accurate, but caters well for the majority use cases.
 
 The module format detection happens in the following order:
-* _System.register / System.registerDynamic_
-  If the source code starts with a number of comments, followed by `System.register` or `System.registerDynamic` as the first line of code.
+* _SystemJS.register / SystemJS.registerDynamic_
+  If the source code starts with a number of comments, followed by `SystemJS.register` or `SystemJS.registerDynamic` as the first line of code.
```
Comment (guybedford) [inline docs/module-formats.md]: System.register, System.registerDynamic

File: docs/production-workflows.md [hunk 6]
```diff
@@ -39,21 +39,21 @@ The bundle extension will automatically download a bundle as soon as an attempt
...
   // and ensures that build/core.js is loaded first
-  System.import('app/app');
+  SystemJS.import('app/app');
   
   // this way a request to any one of 'jquery', 'app/app', 'app/dep', 'lib/third-party'
   // will delegate to the bundle and only a single request is made
 ```
 
-A built file must contain the exact named defines or named `System.register` statements for the modules
+A built file must contain the exact named defines or named `SystemJS.register` statements for the modules
```
Comment (guybedford) [inline docs/production-workflows.md]: ""

File: docs/system-api.md [hunk 7]
```diff
@@ -69,31 +69,31 @@ Promise resolves to the module value.
...
 
-Useful when writing a custom `instantiate` hook or using `System.set`.
+Useful when writing a custom `instantiate` hook or using `SystemJS.set`.
 
-#### System.register([name ,] deps, declare)
+#### SystemJS.register([name ,] deps, declare)
 Type: `Function`
 
-Declaration function for defining modules of the `System.register` polyfill module format.
+Declaration function for defining modules of the `SystemJS.register` polyfill module format.
```
Comment (guybedford) [inline docs/system-api.md]: System.register

File: docs/system-api.md [hunk 8]
```diff
@@ -69,31 +69,31 @@ Promise resolves to the module value.
...
+#### SystemJS.newModule(Object) -> Module
 Type: `Function`
 
 Given a plain JavaScript object, return an equivalent `Module` object.
 
-Useful when writing a custom `instantiate` hook or using `System.set`.
+Useful when writing a custom `instantiate` hook or using `SystemJS.set`.
 
-#### System.register([name ,] deps, declare)
+#### SystemJS.register([name ,] deps, declare)
```
Comment (guybedford) [inline docs/system-api.md]: This is fine as SystemJS though.

File: docs/system-api.md [hunk 9]
```diff
@@ -69,31 +69,31 @@ Promise resolves to the module value.
...
+Declaration function for defining modules of the `SystemJS.register` polyfill module format.
 
 [Read more on the format at the loader polyfill page](https://github.com/ModuleLoader/es6-module-loader/blob/v0.17.0/docs/system-register.md)
 
-#### System.registerDynamic([name ,] deps, executingRequire, declare)
+#### SystemJS.registerDynamic([name ,] deps, executingRequire, declare)
 Type: `Function`
 
-Companion module format to `System.register` for non-ES6 modules.
+Companion module format to `SystemJS.register` for non-ES6 modules.
```
Comment (guybedford) [inline docs/system-api.md]: ""

**Comment by b-strauss** [comment]
@guybedford update based on your feedback

File: docs/system-api.md [hunk 10]
```diff
@@ -69,19 +69,19 @@ Promise resolves to the module value.
...
-#### System.newModule(Object) -> Module
+#### SystemJS.newModule(Object) -> Module
 Type: `Function`
 
 Given a plain JavaScript object, return an equivalent `Module` object.
 
-Useful when writing a custom `instantiate` hook or using `System.set`.
+Useful when writing a custom `instantiate` hook or using `SystemJS.set`.
 
 #### System.register([name ,] deps, declare)
```
Comment (guybedford) [inline docs/system-api.md]: It is actually ok to update this to SystemJS.register for consistency here.

**Comment by guybedford** [comment]
👍  I've added the adjustment manually.

**Pull request merged**